report/docker_run.py

```diff
@@ -0,0 +1,295 @@
+  parser.add_argument(
+      '-l',
+      '--frequency-label',
+      type=str,
+      default=FREQUENCY_LABEL,
+      help=
+      f'Used as part of Cloud Build tags and GCS report directory, default: {FREQUENCY_LABEL}.'
+  )
+  parser.add_argument(
+      '-to',
+      '--run-timeout',
+      type=int,
+      default=RUN_TIMEOUT,
+      help=f'Fuzzing timeout in seconds, default: {RUN_TIMEOUT} seconds.')
+  parser.add_argument(
+      '-sd',
+      '--sub-dir',
+      type=str,
+      default=SUB_DIR,
+      help=
+      f'The subdirectory for the generated report in GCS, default: {SUB_DIR}.')
+  parser.add_argument('-m',
+                      '--model',
+                      type=str,
+                      default=MODEL,
+                      help=f'Large Language Model name, default: {MODEL}.')
+  parser.add_argument(
+      '-d',
+      '--delay',
+      type=int,
+      default=DELAY,
+      help=f'Delay each benchmark experiment by N seconds, default: {DELAY}.')
+  parser.add_argument(
+      '-i',
+      '--local-introspector',
+      type=str,
+      default="false",
+      help=
+      'If set to "true" will use a local version of fuzz introspector\'s webapp'
+  )
+  parser.add_argument(
+      '-ns',
+      '--num-samples',
+      type=int,
+      default=NUM_SAMPLES,
+      help=f'The number of samples to request from LLM, default: {NUM_SAMPLES}')
+  parser.add_argument(
+      '-nf',
+      '--llm-fix-limit',
+      type=int,
+      default=LLM_FIX_LIMIT,
+      help=f'The number of fixes to request from LLM, default: {LLM_FIX_LIMIT}')
+  parser.add_argument(
+      '-vt',
+      '--vary-temperature',
+      type=str,
+      default="true",
+      help=
+      'Use different temperatures for each sample. Set to "false" to disable.')
+  parser.add_argument(
+      '-ag',
+      '--agent',
+      type=str,
+      default="false",
+      help='Enables agent enhancement. Set to "true" to enable.')
+  parser.add_argument('-mr',
+                      '--max-round',
+                      type=int,
+                      default=MAX_ROUND,
+                      help=f'Max trial round for agents, default: {MAX_ROUND}.')
+  parser.add_argument(
+      '-rd',
+      '--redirect-outs',
+      type=str,
+      default="false",
+      help=
+      'Redirects experiments stdout/stderr to file. Set to "true" to enable.')
+
+  args, additional_args = parser.parse_known_args(cmd)
+  args.additional_args = additional_args[
+      1:]  # first element is the separator itself ("--")
+
+  # Parse boolean arguments
+  args.local_introspector = args.local_introspector.lower() == "true"
+  args.vary_temperature = args.vary_temperature.lower() == "true"
+  args.agent = args.agent.lower() == "true"
+  args.redirect_outs = args.redirect_outs.lower() == "true"
+
+  return args
+
+
+def _run_command(command, shell=False):
+  """Runs a command and return its exit code."""
+  if isinstance(command, str) and not shell:
+    command = shlex.split(command)
```

Comment:
could we enforce `command` to be `list[str]` so that we don't have to split and can use subprocess.run directly?